### PR TITLE
[homekit] add setting to block homekit user/pairing deletion

### PR DESCRIPTION
**Pull request by yfre**
there are many reports about lost HomeKit user / pairing information from homekit.json

this PR add setting to block deletion of the user information by binding. the default setting is false which ensures the same behaviour as before this PR.

Signed-off-by: Eugen Freiter freiter@gmx.de

### CODE REVIEW

File: bundles/org.openhab.io.homekit/README.md [hunk 1]
```diff
@@ -98,6 +99,7 @@ org.openhab.homekit:name=openHAB
...
 | port                     | Port under which the HomeKit bridge can be reached.                                                                                                                                                                                     | 9123          |
 | useOHmDNS                | mDNS service is used to advertise openHAB as HomeKit bridge in the network so that HomeKit clients can find it. openHAB has already mDNS service running. This option defines whether the mDNS service of openHAB or a separate service should be used.   | false  |
+| blockUserDeletion        | Blocks HomeKit user deletion in openHAB and as result unpairing of devices. if you experience issue with accessories becoming non-responsive after some time, try to enable this setting. You can also enable this setting if you HomeKit setup is done and you will not re-pair ios devices.                                              | false         |
```
Comment (lolodomo) [inline bundles/org.openhab.io.homekit/README.md:102]: ... If you experience... 
... if your HomeKit...

**Review by lolodomo — CHANGES_REQUESTED** [review]
Just two minor typo in doc

File: bundles/org.openhab.io.homekit/README.md (hunk 1 above)
Comment (yfre) [reply bundles/org.openhab.io.homekit/README.md:102]: @lolodomo two typos in one sentence, argh :( thank you for pointing them out. fixed

**Review by lolodomo — APPROVED** [review]
LGTM

**Comment by yfre** [comment]
@obartelt  try with 
`install https://ci.openhab.org/job/openHAB-Addons/lastSuccessfulBuild/artifact/bundles/org.openhab.io.homekit/target/org.openhab.io.homekit-3.2.0-SNAPSHOT.jar`

if it does not work, then install the stable version of binding via ui and then update with with 
`update <binding_id> https://ci.openhab.org/job/openHAB-Addons/lastSuccessfulBuild/artifact/bundles/org.openhab.io.homekit/target/org.openhab.io.homekit-3.2.0-SNAPSHOT.jar`

**Comment by obartelt** [comment]
With install, it's the same message (I had tried that before, because I thought that it would resolve dependencies when doing it this way, but that doesn't seem to make any difference).

Ah, after installing the 3.1.1 version from the UI, it now has installed the dependencies as well. Strange, as I thought the io.netty packages are so basic that openHAB would need them by itself.

```
267 │ Active │  80 │ 4.1.63.Final          │ Netty/Buffer                                                                                                                                                   
268 │ Active │  80 │ 4.1.63.Final          │ Netty/Codec                                                                                                                                                    
269 │ Active │  80 │ 4.1.63.Final          │ Netty/Codec/HTTP                                                                                                                                               
270 │ Active │  80 │ 4.1.63.Final          │ Netty/Common                                                                                                                                                   
271 │ Active │  80 │ 4.1.63.Final          │ Netty/Handler                                                                                                                                                  
272 │ Active │  80 │ 4.1.63.Final          │ Netty/Resolver                                                                                                                                                 
273 │ Active │  80 │ 4.1.63.Final          │ Netty/Transport                                                                                                                                                
274 │ Active │  80 │ 3.1.1                 │ openHAB Add-ons :: Bundles :: IO :: Homekit
```
The update command did the trick. Thank you! I'm just curious as to why the dependencies aren't automatically installed when doing it via console or addons folder?

**Comment by obartelt** [comment]
Ok, while the update has worked now, I just witnessed something strange concerning the original problem (losing the pairing to the Home app in iOS):

When I restarted my docker container in which openHAB is running, the openHAB bridge completely disappeared from the Home app (I had the devices of the bridge open for debugging purposes, which went completely blank (just a white page with the bridge name on top). Then I went back to the bridge details, which were apparently still there, and then back to the home settings, where the bridge was missing all of a sudden).

Is there maybe a unpairing of sorts happening after all, when openHAB is stopped/restarted? The pairing did reappear and worked after openHAB was running again, but all my other bridges (virtual or real) stay visible inside the Home app when they're not running/connected, and just say that they're not available at the moment.

Maybe this helps. But as I said, the connection still worked after openHAB was up and running again and I even had forgotten to enable the setting of this pull request ;-)

**Comment by obartelt** [comment]
So far no issues with openHAB 3.2 Release. Great job, guys (and girls)!

**Pull request merged**